### DISCUSSION TITLE
Symlink is cached when deploying Laravel Octane

### DISCUSSION THREAD

**Discussion by cerbero90** [deployphp/deployer#2577] [Help needed]
Hi everybody, a standard Laravel app can be run on an NGINX FastCGI server, however an Octane app needs a reverse proxy.

In a FastCGI server we can avoid the cache of the symlink pointing to the new release by using the following directives:

```
fastcgi_param SCRIPT_FILENAME $realpath_root$fastcgi_script_name;
fastcgi_param DOCUMENT_ROOT $realpath_root;
```

That is not possible in an NGINX reverse proxy and the symlink is getting cached, pointing to an old release.

Is there a way to avoid the symlink cache in an NGINX reverse proxy?

Thanks in advance for your help, this is how the NGINX config looks like:
```conf
location @octane  {
    set $suffix "";

    if ($uri = /index.php) {
        set $suffix ?$query_string;
    }

    proxy_http_version 1.1;
    proxy_set_header Host $http_host;
    proxy_set_header Scheme $scheme;
    proxy_set_header SERVER_PORT $server_port;
    proxy_set_header REMOTE_ADDR $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $connection_upgrade;

    proxy_pass http://127.0.0.1:8000$suffix;
}
```

**Comment by Quiss** [comment]
@cerbero90 this is not a deployer problem.
Octane create config in `storage/logs/octane-server-state.json`
and write down the path (ex: `"publicPath": "\/...\/prod\/2408\/public"`)

`artisan octane:reload` can't fix this problem, the only solution i found -> `stop / start octane` (or just `supervisor reload`)

**Reply by antonmedv** [reply]
Reload will stop and restart? Not good. Probably octane needs support for resolving public path.

**Reply by HassanZahirnia** [reply]
@Quiss Do you have any update on this? What's your current solution for this issue, that would provide least impact (downtime) ?

**Reply by Quiss** [reply]
@HassanZahirnia Sorry i missed the notice. When deploy i raise "prev deploy" octane with another port
https://gist.github.com/Quiss/80858d7f67d91913f1b6c788864bf59f

Deploy steps: 
1) Run octane on current version as "deploy mode" && sleep for run worker (3-5 sec in my case)
2) Stop supervisor / Deploy / change symlink / migration / etc.. 
3) Run supervisor & sleep 
4) Stop "deploy octane"

**Reply by Quiss** [reply]
@antonmedv FYI ☝️

**Reply by yusifk88** [reply]
Any fix for this issue ?
